cellxgene_schema_cli/cellxgene_schema/validate.py

```diff
@@ -578,8 +577,31 @@ def _validate_column_feature_is_filtered(self, column: pd.Series, column_name: s
             )
             return
 
+        if self.adata.raw is None:
+            if column.any():
+                self.errors.append(
+                    "'feature_is_filtered' must be False for all features if 'adata.raw' is not present."
+                )
+            return
+        else:
+            sum_X = compute_column_sums(self.adata.X)
+            sum_raw_X = compute_column_sums(self.adata.raw.X)
+            for i, _ in enumerate(sum_X):
+                is_filtered = column[i]
+                raw_column_sum = sum_raw_X[i]
+                # If adata.X has all 0s for the column, then is_filtered must be True or adata.raw.X should also be all 0s
+                if sum_X[i] == 0:
+                    if is_filtered or raw_column_sum == 0:
+                        continue
+                    else:
+                        gene_name = self.adata.var_names[i]
+                        self.errors.append(
```

Comment:
the original ticket ask was for a warning--has that changed, or should we fix here?

also not blocking for the ticket but if it is meant to be a warning only, I was thinking whether it'd be worthwhile to gate this check behind a --warnings flag in the CLI. since warnings aren't really actionable/exposed in the ingestion pipeline, it isn't really useful to run this check in a deployed environment, only locally.

Reply:
i was confused about that as well, but the final schema says we should raise an error if we encounter this issue. i guess the part of the ticket body that says "feature request: raise a warning when X" is actually part of the _context_ for the ticket, not the actual implementation details of the ticket

jason and brian confirmed we want an error here: https://czi-sci.slack.com/archives/C08MF1AJ6F5/p1745937982488189